### PR TITLE
Bugfix/devices mapping

### PR DESCRIPTION
**Pull request by MaksimR-AVISPL**

### CODE REVIEW

File: src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import com.avispl.symphony.api.dal.dto.monitor.ExtendedStatistics;
 import com.avispl.symphony.api.dal.dto.monitor.Statistics;
 import com.avispl.symphony.api.dal.dto.monitor.aggregator.AggregatedDevice;
+import com.avispl.symphony.api.dal.error.CommandFailureException;
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java:12]: Github won't let me comment on things that have not changed, hence commenting here:

1. Please update copyright to 2020-2021 (above)
2. Please add javadoc at class level (below)

File: src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java [hunk 2]
```diff
@@ -429,12 +450,21 @@ private ArrayNode getFirmwareVersions(String deviceModel) {
...
-        authenticate();
-        String devicesResponse = doGet(BASE_URL + "systems", String.class);
-        return objectMapper.readTree(devicesResponse);
+    private JsonNode getSystems(boolean reAuthenticated) throws Exception {
+        try {
+            String devicesResponse = doGet(BASE_URL + "systems", String.class);
+            return objectMapper.readTree(devicesResponse);
+        } catch (FailedLoginException | CommandFailureException fle) {
+            if (reAuthenticated) {
+                throw new FailedLoginException("Failed to get list of systems using SessionID: " + loginId);
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java:463]: who populates loginId?
from what I can gather it was never initialized.

File: src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java [hunk 3]
```diff
@@ -360,14 +361,25 @@ private String retrieveDeviceUrlSegment(String deviceModel){
...
-        String devicesResponse = doGet(BASE_URL + "devices", String.class);
-        JsonNode devices = objectMapper.readTree(devicesResponse);
+    private JsonNode getDevices(boolean reAuthenticated) throws Exception {
+        JsonNode devices;
+        try {
+            String devicesResponse = doGet(BASE_URL + "devices", String.class);
+            devices = objectMapper.readTree(devicesResponse);
+        } catch (FailedLoginException | CommandFailureException fle) {
+            if (reAuthenticated) {
+                throw new FailedLoginException("Failed to get list of devices using SessionID: " + loginId);
```
Comment (ShervinM-AVISPL) [inline src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java:375]: who populates loginId?
from what I can gather it was never initialized.

**Review by ShervinM-AVISPL — CHANGES_REQUESTED** [review]

File: src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java (hunk 2 above)
Comment (MaksimR-AVISPL) [reply src/main/java/com/avispl/symphony/dal/communicator/biamp/sagevue/SageVueCommunicator.java:463]: it is initialized in authenticate() method

**Review by ShervinM-AVISPL — APPROVED** [review]
Approving; please don't forget to update copyright and javadocs at class and constructor levels.

**Pull request merged**